packages/p/pahomqttcpp/xmake.lua

```diff
@@ -0,0 +1,38 @@
+package("pahomqttcpp")
+    set_homepage("https://github.com/eclipse/paho.mqtt.cpp")
+    set_description("Eclipse Paho MQTT C++ Client Library")
+    set_license("EPL v1.0")
+
+    add_urls("https://github.com/eclipse/paho.mqtt.cpp/archive/refs/tags/$(version).zip",
+             "https://github.com/eclipse/paho.mqtt.cpp.git")
+
+    add_versions("v1.2.0", "90c4d8ae4f56bb706120fddcc5937cd0a0360b6f39d5cd5574a5846c0f923473")
+    
+    add_deps("cmake")
+    add_deps("pahomqttc")
+
+    on_install("windows", "linux", "macos", function (package)
+        local pahomqttc = package:dep("pahomqttc")
+        
+        local configs = {"-DPAHO_BUILD_SAMPLES=FALSE",
+                         "-DPAHO_WITH_SSL=FALSE",
+                         "-DPAHO_BUILD_DOCUMENTATION=FALSE"}
+        table.insert(configs, "-DCMAKE_PREFIX_PATH=" .. pahomqttc:installdir())
```

Comment:
please remove it, xmake will add it

Reply:
The build fails on linux for gcc and clang if I remove "-DPAHO_WITH_SSL=FALSE". For some reason, cmake keeps trying to find OpenSSL in that configuration without it.

Reply:
you can disable it first. if you want to improve it to support ssl.

you can patch cmakelists.tx to disable find openssl, and pass `{packagedeps = "openssl"}` to cmake.install to force set their include/lib flags.

you can see other packages.